addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/XMLConnection.java

```diff
@@ -31,55 +36,55 @@
 public class XMLConnection extends AbstractConnection {
     private Logger logger = LoggerFactory.getLogger(XMLConnection.class);
 
-    public final static String XML_GET = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"GET\">";
-    public final static String XML_PUT = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"PUT\">";
-    public final static String XML_END = "</YAMAHA_AV>";
+    private static final String XML_GET = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"GET\">";
+    private static final String XML_PUT = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"PUT\">";
+    private static final String XML_END = "</YAMAHA_AV>";
 
     public XMLConnection(String host) {
         super(host);
     }
 
-    /**
-     * Post the given xml message
-     *
-     * @param message XML formatted message excluding < ?xml > or <YAMAHA_AV> tags.
-     * @throws IOException
-     */
-    @Override
-    public void send(String message) throws IOException {
+    @FunctionalInterface
+    public interface CheckedConsumer<T, R> {
+        R apply(T t) throws IOException;
+    }
+
+    private <T> T postMessage(String prefix, String message, String suffix, CheckedConsumer<HttpURLConnection, T> responseConsumer) throws IOException {
         HttpURLConnection connection = null;
+
         if (message.startsWith("<?xml")) {
             throw new IOException("No preformatted xml allowed!");
         }
-
-        message = XML_PUT + message + XML_END;
+        message = prefix + message + suffix;
 
         writeTraceFile(message);
 
+        URL url = createCrlUrl();
         try {
-            URL url = createCrlUrl();
             connection = (HttpURLConnection) url.openConnection();
             connection.setRequestMethod("POST");
-            connection.setRequestProperty("Content-Length", "" + Integer.toString(message.length()));
+            connection.setRequestProperty("Content-Length", Integer.toString(message.length()));
 
             connection.setUseCaches(false);
             connection.setDoInput(true);
             connection.setDoOutput(true);
 
             // Send request
-            DataOutputStream wr = new DataOutputStream(connection.getOutputStream());
-            try {
+            // ToDo: This should be sent using proper UTF-8 encoding
```

Comment:
I doubt, that Yamaha implemented a proper UTF8 character-set support. The only think that users can provide own strings for, is for the input names?